browser/ui/views/tabs/vertical_tab_strip_browsertest.cc

```diff
@@ -391,3 +441,264 @@ IN_PROC_BROWSER_TEST_F(VerticalTabStripBrowserTest, MAYBE_Fullscreen) {
+    return GetTabStrip(b)->GetDragContext()->IsDragSessionActive();
+  }
+
+  // VerticalTabStripBrowserTest:
+  void SetUpOnMainThread() override {
+    VerticalTabStripBrowserTest::SetUpOnMainThread();
+
+#if defined(USE_AURA)
+    auto* root_window =
+        browser_view()->GetWidget()->GetNativeWindow()->GetRootWindow();
+    ui_controls::InstallUIControlsAura(
+        aura::test::CreateUIControlsAura(root_window->GetHost()));
+
+    auto* widget_delegate_view =
+        browser_view()->vertical_tab_strip_widget_delegate_view_.get();
+    ASSERT_TRUE(widget_delegate_view);
+
+    root_window =
+        widget_delegate_view->GetWidget()->GetNativeWindow()->GetRootWindow();
+    ui_controls::InstallUIControlsAura(
+        aura::test::CreateUIControlsAura(root_window->GetHost()));
+#endif
+
+#if BUILDFLAG(IS_OZONE)
+    // Notifies the platform that test config is needed. For Wayland, for
+    // example, makes it possible to use emulated input.
+    ui::test::EnableTestConfigForPlatformWindows();
+
+    ui::OzonePlatform::InitParams params;
+    params.single_process = true;
+    ui::OzonePlatform::InitializeForUI(params);
+#endif
+
+    ui_controls::EnableUIControls();
+
+    ToggleVerticalTabStrip();
+
+#if BUILDFLAG(IS_WIN)
+    // Sometimes, the window is not activated and it causes flakiness. In order
+    // to make sure the window is the front, do these.
+    browser()->window()->Minimize();
+    browser()->window()->Restore();
+    browser()->window()->Activate();
+#endif
+  }
+};
+
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_WIN)
+// TODO(sko) On Linux test environment, the test doesn't work well
+// TODO(sko) On Windows CI, SendMouse() doesn't work.
+#define MAYBE_DragTabToReorder DISABLED_DragTabToReorder
+#else
+#define MAYBE_DragTabToReorder DragTabToReorder
+#endif
+
+IN_PROC_BROWSER_TEST_F(VerticalTabStripDragAndDropBrowserTest,
+                       MAYBE_DragTabToReorder) {
+  // Pre-conditions ------------------------------------------------------------
+  chrome::AddTabAt(browser(), {}, -1, true);
+
+  auto* widget_delegate_view =
+      browser_view()->vertical_tab_strip_widget_delegate_view_.get();
+  ASSERT_TRUE(widget_delegate_view);
+
+  auto* region_view = widget_delegate_view->vertical_tab_strip_region_view();
+  ASSERT_TRUE(region_view);
+  ASSERT_EQ(VerticalTabStripRegionView::State::kExpanded, region_view->state());
+
+  // Drag and drop a tab to reorder it -----------------------------------------
+  GetTabStrip(browser())->StopAnimating(
+      /* layout= */ true);  // Drag-and-drop doesn't start when animation is
+                            // running.
+  PressTabAt(browser(), 0);
+  MoveMouseTo(GetCenterPointInScreen(GetTabAt(browser(), 1)));
+  EXPECT_TRUE(IsDraggingTabStrip(browser()));
+  ReleaseMouse();
+  {
+    // Regression test for https://github.com/brave/brave-browser/issues/28488
+    // Check if the tab is positioned properly after drag-and-drop.
+    auto* moved_tab = GetTabAt(browser(), 1);
```

Comment:
Suggestion: How about doing one more check here -  original tab at zero is equal to this `moved_tab` after releasing is done?

Reply:
That's good idea!